### DISCUSSION TITLE
Compatibility with HP Designjet Color 3D

### DISCUSSION THREAD

**Discussion by rickreiniers** [mayrthom/CartridgeWriter_uPrint#2] [Q&A] [answered]
Is there a way to make the software compatible with the HP designjet Color 3D too? It is basically a stratasys Uprint. Only both machine Settings don't seem to match. 
I just don't know how to get the machine ID to rebuild the software myself with this option added.

**Comment by mayrthom** [comment, marked as answer]
Thank you for opening the discussion. 

There was a bug with the general decoding of the serial code, which I fixed right now.

Please try it again with both machine settings.

Let me know, if it works now.

**Reply by rickreiniers** [reply]
The error is gone, but now if i hit the decrypt button the software crashes with Uprint / Uprint Plus selected.
With Uprint SE selected it shows a checksum error.

**Reply by mayrthom** [reply]
Then I guess the problem is that the Designjet 3D has a different machine id.

Can you copy please the received hex code from the printer into your answer?

**Reply by rickreiniers** [reply]
I have found the error by running the software in visual studio:
It has an unhandled exception in this part:
public static Material FromId(double id) { return materials.Where(m => m.Id.Equals(id)).First(); }
So the material ID seems to be the problem.

This the received HEX code:

er 0 0 0 128
Model carrier ID (8 8-bit values)
000000: b3 61 67 0b d0 14 10 b7                           .ag.....

Model carrier in bay 0: 128 bytes at address 0 (128 8-bit values)
000000: 11 22 52 9b 92 49 f0 6d 14 7f 67 9e 49 cb a9 2d   ."R..I.m..g.I..-
000016: db 79 a3 92 b8 08 15 e6 46 6a be 24 35 cc 71 56   .y......Fj.$5.qV
000032: 26 60 a0 e6 81 3f e0 ad b8 68 f7 40 a9 7b a0 9e   &`...?...h.@.{..
000048: b8 68 f7 40 a9 7b a0 9e b0 be 0b 48 08 c6 c1 1c   .h.@.{.....H....
000064: 38 8a 00 00 00 00 75 a6 55 aa 55 2b c3 03 4a be   8.....u.U.U+..J.
000080: 9c 5e 00 00 00 00 00 00 8a 53 de 8d ba 11 18 4e   .^.......S.....N
000096: 2a 92 c6 ad fc a9 f1 d2 53 54 52 41 54 41 53 59   *.......STRATASY
000112: 53 c3 bf b9 62 68 de e8 55 95 f2 1a bf 6f 97 42   S...bh..U....o.B

**Reply by mayrthom** [reply]
Thank you for the hex code.
You are right, it is the material id!

It looks like that the HP Designjet is using some weird Material-IDs, different to the Material-IDs of the Stratasys. So they are not in the material list.
It would be possible to add each unknown Material-ID and it's name to the Material list, but i changed the program a bit, so it should work with any material ID now. If it is not in the material list, it won't crash anymore, it will just show that it is an other material which is not in the list and also its hex code.

Please try it out and tell me if it works!

**Reply by rickreiniers** [reply]
I have some program which looks the same, it was capable of rewriting the support chip, but it fails on the material. So i'm going to try the new method. 
If i do know the correct hex codes i can add them to the material list if i'm correct?

**Reply by mayrthom** [reply]
Which program that looks the same do you mean?

I just updated this version right now and I think it should work right out of the box with the designjet.
But if you want to, you can add the material-IDs and the name to the list, then it looks neater when the correct name is displayed.

**Comment by rickreiniers** [comment]
The program i mentioned was from some forum but only an executable file. 

Thanks for all the effort.
It does work fine now. I was able to refill the material as well now. 

The support material is called SR30L on the label. It is listed as 0xd3
The ABS dark grey material is called P430L. It is listed as 0xe3

Since HP doesn't deliver material for this printer anymore this is one of the solutions to keep it operational. 

Do you have any suggestions on usable material for the ABS or Support?

**Reply by mayrthom** [reply]
No problem, glad that it works now.

I added the two materials to the material list now.

It's been some years that I used the stratasys the last time, so I can't tell you any specific material, but we didn't use any special filament. We tried different materials, that we could order at our local electronic warehouse here in Austria and also the cheap stuff worked just fine. For PLA there's absolutely no chance to get it working.
One time the print detached from the printing plate ans was pushed into the printhead, which messed up the printhead quiet. Maybe the diameter was too thin, so that it didn't stick that well.
We hadn't any substitute for the support, so we designed everything so, that it worked nearly without any support.

**Reply by mayrthom** [reply]
But I think the original Stratasys material for the uPrint should also work for the HP Design 3D, but it's quite expensive.

**Reply by rickreiniers** [reply]
For the support i will order the original anyway, i was just curious how people are using abs with a different temperature range, since the temperature is around 300 degrees C.
I don’t have the MaracaEX software to alter the temperature.
